[DOCS] Opster Migration: Add errors to troubleshooting pages (#1116)
elastic/docs-content: troubleshoot/elasticsearch/unable-to-retrieve-node-fs-stats.md

---
applies_to:
  stack: 
  deployment:
    eck: 
    ess: 
    ece: 
    self: 

> navigation_title: "Error: unable to retrieve node fs stats"

Reviewer: Still need to make this capitalization change

To resolve this

1. Ensure that the nodes are up and running
2. Check the network connectivity between the client and the nodes
3. Verify the configuration settings. 

If the issue persists, consider checking the {{es}} logs for more detailed error information.

## What it means

This log message typically comes from {{kib}} when it tries to connect to {{es}} during its startup routine. {{kib}} acts as a client to {{es}} and requires access to:

- The cluster’s host and port
- Authentication credentials (if required)
- TLS settings (if applicable)

If {{kib}} can’t reach the nodes listed in its configuration, it can’t determine if the versions are compatible, and it throws this error.

## Potential causes

- One or more entries in `elasticsearch.hosts` are unreachable or misconfigured
- The `KBN_PATH_CONF` environment variable points to a different config file
- A firewall is blocking access between {{kib}} and {{es}}

## Configuration locations

Settings are defined in `kibana.yml`, usually located at `$KIBANA_HOME/config`. You can override its path with:

> ```console

Reviewer: ```suggestion
```bash
```

KBN_PATH_CONF=/home/kibana/config ./bin/kibana
```

Relevant settings:

```yaml
elasticsearch.hosts: ["http://localhost:9200"]
elasticsearch.username: "kibana"
elasticsearch.password: "your_password"
elasticsearch.ssl.certificateAuthorities: ["path/to/ca.crt"]
```
{{kib}} tries every endpoint in `elasticsearch.hosts`, so even one unreachable node can cause the log. Use `https` if your cluster requires encrypted communication.

## How to resolve it

### Test connectivity

Try to connect to each host in `elasticsearch.hosts` using `curl`:

```console
curl http://es01:9200/
```

> If using TLS:

Reviewer: ```suggestion
If you're using TLS, try one of the following:
```

> ```console

Reviewer: ```suggestion
```bash
```

Reviewer: (doesn't render correctly with `console`)

# Insecure test
curl -u elastic -k https://es01:9200/

# Secure test
curl -u elastic --cacert ~/certs/ca/ca.crt https://es01:9200/
```

> You should see a response like:

Reviewer: ```suggestion
Example response:
```

> ```console-result

Reviewer: ```suggestion
```json
```

{
  "name" : "node01",
  "cluster_name" : "elasticsearch",
  "cluster_uuid" : "fxP-R0FTRcmTl_AWs7-DiA",
  "version" : {
    "number" : "7.13.3",
    "build_flavor" : "default",
    "build_type" : "tar",
    "build_hash" : "5d21bea28db1e89ecc1f66311ebdec9dc3aa7d64",
    "build_date" : "2021-07-02T12:06:10.804015202Z",
    "build_snapshot" : false,
    "lucene_version" : "8.8.2"
  },
  "tagline" : "You Know, for Search"
}
```

> If all else fails, refer to your {{kib}} logs for more details and context.

Reviewer: ```suggestion
If you're still encountering issues, check the {{kib}} logs for more details and context.
```
